[error-tracking] Add documentation for link issues to PRs (#35369)
DataDog/documentation: content/en/error_tracking/link_pull_requests.md

---
title: Link Pull Requests
further_reading:
  - link: '/error_tracking/issue_states/'
    tag: 'Documentation'
    text: 'Issue States in Error Tracking'
  - link: '/error_tracking/suspect_commits/'
    tag: 'Documentation'
    text: 'Suspect Commits'
  - link: '/integrations/guide/source-code-integration'
    tag: 'Documentation'
    text: 'Source Code Integration'
---

## Overview

Link GitHub pull requests to Error Tracking issues to track the full lifecycle of an error, from detection to resolution. After you paste an Error Tracking issue URL in a pull request description, Datadog automatically links the PR to the issue and resolves the issue when the PR is merged.

With linked pull requests, you can:

- **Track issue resolution**: See which pull requests are addressing an Error Tracking issue directly from the issue panel.
- **Automatically resolve issues**: When a linked pull request is merged, Datadog automatically moves the issue to **RESOLVED** status.

> 

Author: done in https://github.com/DataDog/documentation/pull/35369/commits/2de7eeef9140a5a035049f8a064e3079b2e99f1f

also moved the setup (integration, SCI) in there

## Link a pull request to an issue

To link a pull request to an Error Tracking issue:

1. In [Error Tracking][1], open the issue you want to link.

> 1. Copy the issue URL from your browser's address bar.

Author: added in https://github.com/DataDog/documentation/pull/35369/commits/e887010c41e4e112457aa433178e313b0bbe1e7f

<div class="alert alert-info">Updating a pull request description to add or remove issue URLs automatically updates the link.</div>

### View linked pull requests

To view linked pull requests:
1. Navigate to [Datadog Error Tracking][1].
1. Select the issue you want to link to open its side panel.
1. In the top right corner, click the **Actions** dropdown to see the linked PR under **PULL REQUESTS**.
{{< img src="error_tracking/pull-request-find.png" alt="View pull requests linked to an Error Tracking issue" style="height:300px;" >}}

### Unlink a pull request

To unlink a pull request from an issue, open the GitHub pull request and remove the Error Tracking issue URL from the description.

## Automatic state changes

Linked pull requests trigger automatic issue state transitions: 

| Initial state | Action | Resulting state |
|---|---|---|
| Issue is `For Review` | A PR is linked to the issue | Issue moves to `Reviewed` |
| Issue is `Reviewed` | A linked PR is merged | Issue moves to `Resolved` |
| Issue is `Resolved` | A linked PR is reopened | Issue moves to `Reviewed` |
| Issue is `Resolved` | A new open PR is linked to the issue | Issue moves to `Reviewed` |

> <div class="alert alert-info">If an issue is linked to multiple pull requests, the issue moves to `Resolved` only when the last open (non-closed, non-merged) PR is merged. <br><br> Closing a PR without merging does not resolve the issue.</div>

Reviewer: ```suggestion
<div class="alert alert-info">If an issue is linked to multiple pull requests, the issue moves to <code>Resolved</code> only when the last open (non-closed, non-merged) PR is merged. <br><br> Closing a PR without merging does not resolve the issue.</div>
```

    text: 'Source Code Integration'
---

## Overview

Link GitHub pull requests to Error Tracking issues to track the full lifecycle of an error, from detection to resolution. After you paste an Error Tracking issue URL in a pull request description, Datadog automatically links the PR to the issue and resolves the issue when the PR is merged.

With linked pull requests, you can:

- **Track issue resolution**: See which pull requests are addressing an Error Tracking issue directly from the issue panel.
- **Automatically resolve issues**: When a linked pull request is merged, Datadog automatically moves the issue to **RESOLVED** status.

## Prerequisites

Install [the GitHub integration][2] with read permissions for pull requests.

Enable [Source Code Integration][3] so that Datadog can process GitHub webhooks and link pull requests to Error Tracking issues.

You need the following [permission][1] to link your PRs to Error Tracking issues:

- Error Tracking Issue Write

## Link a pull request to an issue

To link a pull request to an Error Tracking issue:

1. In [Error Tracking][1], open the issue you want to link.
1. Copy the issue URL from the Actions dropdown of the issue panel.

> 1. Paste the issue URL into the description of your GitHub pull request.

Reviewer: ```suggestion
1. Navigate to [Datadog Error Tracking][1] and select the issue you want to link to open its side panel.
1. Click the **Action** dropdown in the top right corner, then select **Link a Pull Request** to copy the issue URL.
1. Open your Github pull request and paste the issue URL into the description.
```

## View linked pull requests

Linked pull requests appear in the Error Tracking issue panel.

To view linked pull requests:

1. In [Error Tracking][1], open an issue.
1. Linked pull requests are displayed in the Actions dropdown of the issue panel.

{{< img src="error_tracking/pull-request-find.png" alt="View pull requests linked to an Error Tracking issue" style="height:300px;" >}}

## Unlink a pull request

To unlink a pull request from an issue, edit the pull request description and remove the Error Tracking issue URL.

## Automatic state changes

Linked pull requests trigger automatic issue state transitions. If you link or merge a pull request, the linked Error Tracking issue state is updated accordingly:

| Initial state | Action | Resulting state |
|---|---|---|
| Issue is `For Review` | A PR is linked to the issue | Issue moves to `Reviewed` |
| Issue is `Reviewed` | A linked PR is merged | Issue moves to `Resolved` |
| Issue is `Resolved` | A linked PR is reopened | Issue moves to `Reviewed` |
| Issue is `Resolved` | A new open PR is linked to the issue | Issue moves to `Reviewed` |

> If an issue is linked to multiple pull requests, the issue moves to `Resolved` only when the last open (non-closed, non-merged) PR is merged. Closing a PR without merging does not resolve the issue.

Reviewer: ```suggestion
<div class="alert alert-info">If an issue is linked to multiple pull requests, the issue moves to `Resolved` only when the last open (non-closed, non-merged) PR is merged. <br><br> Closing a PR without merging does not resolve the issue.</div>
```
We may want this to pop-up a bit more.
